app/models/papi_client.rb

```diff
@@ -324,6 +351,54 @@ def get_command_line_options(study_file, action)
     opts
   end
 
+  # get command line arguments for differential expression runs
+  #
+  # * *params*
+  #   - +options+ (Hash) => Hash of DE-specific parameters, whose names must include all values from DE_PARAMETER_NAMES
+  #
+  # * *returns*
+  #   - (Array) => Array of DE-specific command line arguments to pass to Docker in "exec" form
+  #
+  # * *raises*
+  #   - (ArgumentError) => if any required parameters from DE_PARAMETER_NAMES are missing or malformed
+  def get_de_parameters(options)
```

Comment:
yeah, if we have a `DifferentialExpressionParams` class, this method will get much simpler if it gets to deal with an object with fields rather than having to match keys/names with `options.detect`. That `DifferentialExpressionParams` class can have it's own `valid?` or `validate` method, which will have the advantage of moving the DE-specific validation logic into one place, and this method can just focus on python command line generation